test/e2e/util_gatewayapi_test.go

```diff
@@ -741,121 +698,88 @@ func assertCatalogSource(t *testing.T, namespace, csName string) error {
-			if pending || unready {
-				t.Logf("found ServiceMeshControlPlane %s/%s, but it isn't ready. Retrying...", smcp.Namespace, smcp.Name)
-				return false, nil
-			}
-			if len(smcp.Status.Readiness.Components["ready"]) > 0 {
-				t.Logf("found ServiceMeshControlPlane %s/%s with ready components: %v", smcp.Namespace, smcp.Name, smcp.Status.Readiness.Components["ready"])
-				return true, nil
-			}
+		if istio.Status.GetCondition(sailv1.IstioConditionReady).Status == metav1.ConditionTrue {
+			t.Logf("Found Istio %s/%s, and it reports ready.", istio.Namespace, istio.Name)
+			return true, nil
 		}
-		t.Logf("found ServiceMeshControlPlane %s/%s but could not determine its readiness. Retrying...", smcp.Namespace, smcp.Name)
+		t.Logf("Found Istio %s/%s, but it isn't ready.  Retrying...", istio.Namespace, istio.Name)
 		return false, nil
 	})
 	return err
 }
 
-// deleteExistingSMCP deletes if the SMCP exists and returns an error if not.
-func deleteExistingSMCP(t *testing.T) error {
+// deleteExistingIstio deletes if the Istio exists and returns an error if not.
+func deleteExistingIstio(t *testing.T) error {
 	t.Helper()
-	existingSMCP := &maistrav2.ServiceMeshControlPlane{}
-	newSMCP := &maistrav2.ServiceMeshControlPlane{}
-	nsName := types.NamespacedName{Namespace: operatorcontroller.DefaultOperandNamespace, Name: openshiftSMCPName}
+	existingIstio := &sailv1.Istio{}
+	newIstio := &sailv1.Istio{}
+	nsName := types.NamespacedName{Namespace: operatorcontroller.DefaultOperandNamespace, Name: openshiftIstioName}
 
 	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, false, func(context context.Context) (bool, error) {
-		if err := kclient.Get(context, nsName, existingSMCP); err != nil {
-			t.Logf("failed to get smcp %s: %v", nsName.Name, err)
+		if err := kclient.Get(context, nsName, existingIstio); err != nil {
+			t.Logf("Failed to get Istio %s: %v", nsName.Name, err)
 			return false, nil
 		}
 		return true, nil
 	})
 	if err != nil {
-		return fmt.Errorf("timed out getting smcp %s: %w", nsName.Name, err)
+		return fmt.Errorf("timed out getting Istio %s: %w", nsName.Name, err)
 	}
-	// deleting SMCP.
-	err = kclient.Delete(context.Background(), existingSMCP)
+	t.Logf("Deleting Istio %s...", existingIstio.Name)
+	err = kclient.Delete(context.Background(), existingIstio)
 	if err != nil {
-		t.Errorf("failed to delete smcp %s: %v", nsName.Name, err)
+		t.Errorf("Failed to delete Istio %s: %v", nsName.Name, err)
 		return err
 	}
 	err = wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 1*time.Minute, false, func(ctx context.Context) (bool, error) {
-		if err := kclient.Get(ctx, nsName, newSMCP); err != nil {
+		if err := kclient.Get(ctx, nsName, newIstio); err != nil {
 			if kerrors.IsNotFound(err) {
 				return true, nil
 			}
-			t.Logf("failed to get deleted SMCP %s: %v", nsName.Name, err)
+			t.Logf("Failed to get deleted Istio %s: %v", nsName.Name, err)
```

Comment:
Is something like this more accurate?
```suggestion
			t.Logf("Deleted Istio %s not yet recreated: %v", nsName.Name, err)
```

Reply:
Maybe something like, "Unexpected error when trying to get Istio after deletion"?

Reply:
That would work.

Reply:
Done in https://github.com/openshift/cluster-ingress-operator/compare/8055a976678a3377646399d0110f8cbe4d0e9144..25dde982b0f91fe5bed88fc44da904d920f03d2a.